training/MESSAGING_OVERVIEW.md

```diff
@@ -0,0 +1,138 @@
+
+To break this request down, `cmd` is the string name that maps to the function you want to use in Chapel. In most cases the string name will be almost identical to the Chapel function name to keep things easy to maintain. `args` is a JSON of all the arguments being passed to the server for processing. 
+
+<a id="receiveReq"></a>
+## Receiving a Request on the Server
+
+As mentioned above, the Command Map handles mapping the given `cmd` string to the proper function in the Chapel based on the available Chapel Modules. In order for a function within a module to be found by the Command Map it must first be registered. An example of this, using the request in the previous section as an example, would be as follows:
+
+`registerFunction("serverFunction", serverFunctionMsg, getModuleName());`
+
+The three components of the `registerFunction` method are the `cmd` string, the Chapel function name, and the Module name which in most cases we use `getModuleName()` to get the name of the module the function is called from.
+
+All messages sent to the server are in JSON format. So before any arguments sent from the client can be used, they must first be parsed. To do this use the function `parseMessageArgs(payload, argSize)`. This will parse your message arguments into a Map that can then be accessed using accessor methods. The most common of these accessor methods is `getValueOf(argName)` which will return the value of a argument that matches the passed in name.
+
+For example, assuming the server request in the above section is what is being received, there are two arguments in the JSON, `arg1` and `arg2`. To access these arguments in Chapel, we will parse the message, or `payload`, into a variable called `msgArgs`. Then using the `getValueOf()` accessor method, we will set variables `arg1` and `arg2` equal to the values of their corresponding JSON arguments.
+
+```chapel
+var msgArgs = parseMessageArgs(payload, argSize);
+var arg1 = msgArgs.getValueOf("arg1");
+var arg2 = msgArgs.getValueOf("arg2");
+```
+
+Now the values of `arg1` and `arg2` are available for use within this Chapel function.
+
+<a id="newModule"></a>
+## Creating a New Chapel Module
+
+Adding a new Chapel module is simple. Create a file of the same name as your Chapel module. If there is no communication between Client and Server in your module, there's nothing special needed! Otherwise, there are two important things to remember when creating a new module that have communication between Client and Server:
+
+- Add the module name to `serverModules.cfg`. (This file sits at the root level of the Arkouda repo).
+- At the end of your Chapel module, include `use CommandMap;` followed by the `registerFunction(stringName, functionName, moduleName)` method for each function in your module that will receive a request from the Client.

```

Comment:
I would break the stuff on the map out into a code block so it reads more like the code. Maybe something like 

At the end of your Chapel module, you will need to pull in CommandMap and register any message processing functions. 
```chapel
use CommandMap;
registerFunction(stringName, functionName, moduleName)
```